app/models/invalid_date.rb

```diff
@@ -0,0 +1,9 @@
+class InvalidDate
```

Comment:
This might be better inlined in the form as it's an internal detail of how the form works.

```ruby
class PersonalDetailsForm < Form
  class InvalidDate
```
What do you think about using a struct for this? Would it work?
```ruby
InvalidDate = Struct.new(:day, :month, :year, keyword_init: true) do
  def future?
    false
   end
end
```

Reply:
ah this is how i had it an hour ago! agreed 👍